core/config_test.go

```diff
@@ -0,0 +1,28 @@
+package core
+
+import "testing"
+
+func TestNormalizeURL(t *testing.T) {
+	cases := map[string]string{
+		"https://hsm.chain.com":     "https://hsm.chain.com",
+		"https://hsm.chain.com:443": "https://hsm.chain.com",
+		"https://hsm.chain.com:80":  "https://hsm.chain.com:80",
+		"https://hsm.chain.com/":    "https://hsm.chain.com",
+		"https://hsm.chain.com:":    "https://hsm.chain.com",
+		"HTTPS://HSM.CHAIN.COM":     "https://hsm.chain.com",
+		"http://hsm.chain.com:80":   "http://hsm.chain.com",
+		"http://hsm.chain.com:443":  "http://hsm.chain.com:443",
+	}
+
+	for u, want := range cases {
+		t.Run(u, func(t *testing.T) {
+			normalized, err := normalizeURL(u)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if normalized != want {
+				t.Errorf("got %s, want %s", normalized, want)
```

Comment:
This might be a nice time to write

```go
t.Errorf("normalizeURL(%q) = %q, want %s", u, normalized, want)
```

and then maybe we don't even need a subtest, idk.